lib/active_model/serializer/adapter/json.rb

```diff
@@ -4,44 +4,122 @@ class ActiveModel::Serializer::Adapter::Json < ActiveModel::Serializer::Adapter
 
         def serializable_hash(options = nil)
           options ||= {}
+          serialized_hash = {}
+
           if serializer.respond_to?(:each)
-            result = serializer.map { |s| FlattenJson.new(s).serializable_hash(options) }
+            serialized_hash = serialize_array_without_root(serializer, options)
           else
-            hash = {}
+            serialized_hash = resource_object_for(serializer)
+            serialized_hash = add_resource_relationships(serialized_hash, serializer)
```

Comment:
I actually think that's a bit of an anti-pattern. 

This way, each method is it's own individual that can be tested, and doesn't rely on the results of others.

Maybe think of it this way?
both methods return a hash that can be serialized. 
but the second method adds *additional* data, that otherwise wouldn't be required for the resource we are trying preparing to serialize.